lnrpc/invoicesrpc/utils.go

```diff
@@ -11,15 +12,42 @@ import (
 	"github.com/lightningnetwork/lnd/zpay32"
 )
 
+// decodePayReq decodes the invoice payment request if present. This is needed,
+// because not all information is stored in dedicated invoice fields. If there
+// is no payment request present, a dummy request will be returned. This can
+// happen with very old invoices and key send.
```

Comment:
what is it that can happen with old invoices? is it possible for an old invoice to have a zero length payment request, or something else?

Reply:
I went back in history and found the introduction of the invoice payment request field in PR https://github.com/lightningnetwork/lnd/pull/317. From the change set I get that migrations weren't done yet at that point. The subsequent release https://github.com/lightningnetwork/lnd/releases/tag/v0.4-beta also describes this.

My conclusion is therefore that we can assume that the payment request is always present.

Comment updated.